cache-for-performance/service-worker.js

```diff
@@ -0,0 +1,74 @@
+// [Working example](/serviceworker-cookbook/offline-status/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+// Files required to make this app work offline
+var REQUIRED_FILES = [
+  'random-1.png',
+  'random-2.png',
+  'index.html',
+  'index.js'
+];
+
+// Non-essential files which may be used often and so
+// it would be benefitial to add to SW cache for sake of speed/performance
+var PERFORMANCE_FILES = [
+  'random-3.png',
+  'random-4.png',
+  'random-5.png',
+  'random-6.png'
+];
+
+self.addEventListener('install', function(event) {
+  // Perform install step:  loading each required file into cache
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        // Add non-essential files to cache if possible
+        // Since this doesn't "return", if any item fails, the
+        // service worker still installs properly with
+        // essential offline files
+        cache.addAll(PERFORMANCE_FILES);
```

Comment:
The main point of this demo is using addAll but not returning it, so it cannot interfere with required core files.